include/z64camera.h

```diff
@@ -54,22 +54,22 @@
 //! there is no cam value mapped to `HUD_VISIBILITY_NOTHING_INSTANT`.
 //! @note: since 0 means `HUD_VISIBILITY_ALL`,
 //! there is no cam value mapped to `HUD_VISIBILITY_NO_CHANGE`.
-#define CAM_HUD_VISIBILITY_ALL                          CAM_HUD_VISIBILITY(0) // HUD_VISIBILITY_ALL
-#define CAM_HUD_VISIBILITY_NOTHING                      CAM_HUD_VISIBILITY(HUD_VISIBILITY_NOTHING)
-#define CAM_HUD_VISIBILITY_NOTHING_ALT                  CAM_HUD_VISIBILITY(HUD_VISIBILITY_NOTHING_ALT)
-#define CAM_HUD_VISIBILITY_HEARTS_FORCE                 CAM_HUD_VISIBILITY(HUD_VISIBILITY_HEARTS_FORCE)
-#define CAM_HUD_VISIBILITY_A                            CAM_HUD_VISIBILITY(HUD_VISIBILITY_A)
-#define CAM_HUD_VISIBILITY_A_HEARTS_MAGIC_FORCE         CAM_HUD_VISIBILITY(HUD_VISIBILITY_A_HEARTS_MAGIC_FORCE)
-#define CAM_HUD_VISIBILITY_A_HEARTS_MAGIC_MINIMAP_FORCE CAM_HUD_VISIBILITY(HUD_VISIBILITY_A_HEARTS_MAGIC_MINIMAP_FORCE)
-#define CAM_HUD_VISIBILITY_ALL_NO_MINIMAP_BY_BTN_STATUS CAM_HUD_VISIBILITY(HUD_VISIBILITY_ALL_NO_MINIMAP_BY_BTN_STATUS)
-#define CAM_HUD_VISIBILITY_B                            CAM_HUD_VISIBILITY(HUD_VISIBILITY_B)
-#define CAM_HUD_VISIBILITY_HEARTS_MAGIC                 CAM_HUD_VISIBILITY(HUD_VISIBILITY_HEARTS_MAGIC)
-#define CAM_HUD_VISIBILITY_B_ALT                        CAM_HUD_VISIBILITY(HUD_VISIBILITY_B_ALT)
-#define CAM_HUD_VISIBILITY_HEARTS                       CAM_HUD_VISIBILITY(HUD_VISIBILITY_HEARTS)
-#define CAM_HUD_VISIBILITY_A_B_MINIMAP                  CAM_HUD_VISIBILITY(HUD_VISIBILITY_A_B_MINIMAP)
-#define CAM_HUD_VISIBILITY_HEARTS_MAGIC_FORCE           CAM_HUD_VISIBILITY(HUD_VISIBILITY_HEARTS_MAGIC_FORCE)
+#define CAM_HUD_VISIBILITY_ALL                          (0) // HUD_VISIBILITY_ALL
+#define CAM_HUD_VISIBILITY_NOTHING                      (HUD_VISIBILITY_NOTHING)
+#define CAM_HUD_VISIBILITY_NOTHING_ALT                  (HUD_VISIBILITY_NOTHING_ALT)
+#define CAM_HUD_VISIBILITY_HEARTS_FORCE                 (HUD_VISIBILITY_HEARTS_FORCE)
+#define CAM_HUD_VISIBILITY_A                            (HUD_VISIBILITY_A)
+#define CAM_HUD_VISIBILITY_A_HEARTS_MAGIC_FORCE         (HUD_VISIBILITY_A_HEARTS_MAGIC_FORCE)
+#define CAM_HUD_VISIBILITY_A_HEARTS_MAGIC_MINIMAP_FORCE (HUD_VISIBILITY_A_HEARTS_MAGIC_MINIMAP_FORCE)
+#define CAM_HUD_VISIBILITY_ALL_NO_MINIMAP_BY_BTN_STATUS (HUD_VISIBILITY_ALL_NO_MINIMAP_BY_BTN_STATUS)
+#define CAM_HUD_VISIBILITY_B                            (HUD_VISIBILITY_B)
+#define CAM_HUD_VISIBILITY_HEARTS_MAGIC                 (HUD_VISIBILITY_HEARTS_MAGIC)
+#define CAM_HUD_VISIBILITY_B_ALT                        (HUD_VISIBILITY_B_ALT)
+#define CAM_HUD_VISIBILITY_HEARTS                       (HUD_VISIBILITY_HEARTS)
+#define CAM_HUD_VISIBILITY_A_B_MINIMAP                  (HUD_VISIBILITY_A_B_MINIMAP)
+#define CAM_HUD_VISIBILITY_HEARTS_MAGIC_FORCE           (HUD_VISIBILITY_HEARTS_MAGIC_FORCE)
```

Comment:
Im confused about this change. are these defines doing anything anymore? they seem like just aliases and you may as well use the actual define

Reply:
See the comment chain https://github.com/zeldaret/oot/pull/1225#discussion_r1038635452

In particular this single comment: https://github.com/zeldaret/oot/pull/1225#discussion_r1038820849

> So to summarize the challenge:
> 
> For OoT: the 3 special values that don't map 1-to-1 are:
> `HUD_VISIBILITY_ALL = 50` which maps to `0` for cam-hudVisibility
> `HUD_VISIBILITY_NOTHING_INSTANT = 52` which has no mapping for cam hudVisibility (interpreted as `HUD_VISIBILITY_A`)
> `HUD_VISIBILITY_NO_CHANGE = 0` which has no mapping for cam hudVisibility (interpreted as `HUD_VISIBILITY_ALL`)
> 
> And then there's a cam hudVisibility of `0xF` which has no mapping to a hudVisibility (but is roughly the same behavior as `HUD_VISIBILITY_NO_CHANGE`) by skipping over the camera code of processing cam-hudVisibility
> 
> In MM, there's all the above plus enum values `0xF-0x16` which all have no mapping for cam hudVisibility, although that seems more of a result of just hacking on more cases after-the-fact without taking cam-hudVisibility into account (+1 for MM being just an OoT RomHack)

Reply:
right, but before they were being passed into another macro `CAM_HUD_VISIBILITY(HUD_VISIBILITY_A)` which was doing the proper shifting and stuff.
With this new change that layer is now gone and its just a name change, which now makes it seem pointless to me

Reply:
While unfortunate, there's two separate reasons for having the defines this way.

1) The change from having the defines in the upper bits (e.g. `0x0100`) to having the defines in the lower bits (e.g. `0x0001`) is because the `initField` stores these variables over the mask: `0x000F`, so having the defines as they were previously (`0x0100`) means these macros can't be used with the `initField`, even though the `initField` uses these values. So by using the lower-shift version, this allows them to be used for both the OnePoint `initField` and for the `CAM_INTERFACE_FIELD` macro

2) While now it seems like a 1-to-1 mapping, using the `HUD_VISIBILITY` macros would lead to unexpected cases if you were to use them. Using `HUD_VISIBILITY_B` would work just fine, but using `HUD_VISIBILITY_NO_CHANGE` or `HUD_VISIBILITY_ALL` would have completely unexpected results. The issue with using `HUD_VISIBILITY_` directly is summarized here:
https://github.com/zeldaret/oot/pull/1225#discussion_r1038820849

Reply:
The motivation from the beginning wasn't about shifts, it was about some most but not all uses of `HUD_VISIBILITY_` mapping 1-to-1. i.e. if we expect modders to use `HUD_VISIBILITY_`, then they will run into issues using `HUD_VISIBILITY_ALL`, `HUD_VISIBILITY_NOTHING_INSTANT`, and `HUD_VISIBILITY_NO_CHANGE` (and will run into even more changes in MM, although OnePoint is removed from MM so that's something I guess)

Reply:
The comment linked above is good to reference, but for example, using the `HUD_VISIBILITY_NOTHING_INSTANT` in this macro would lead to the completely unexpected result of being interpreted as `HUD_VISIBILITY_A`, while using the macro `HUD_VISIBILITY_NO_CHANGE` would lead to the opposite effect of being interpreted as `HUD_VISIBILITY_ALL`

Reply:
I see
I find the wrapper defines just as confusing as the system not supporting all modes equally.
But maybe the comment above can be changed to be more friendly. as someone not familiar with the system, this doesnt really mean that much to me
```
//! @note: since `interfaceField` can only have `0 - 0xF` values,
//! there is no cam value mapped to `HUD_VISIBILITY_NOTHING_INSTANT`.
//! @note: since 0 means `HUD_VISIBILITY_ALL`,
//! there is no cam value mapped to `HUD_VISIBILITY_NO_CHANGE`.
```

something like
```
These defines exist to clarify exactly which HUD Visibility modes are supported by the camera system. While most of them map 1 to 1 with their HUD visibility counterparts, not all values for HUD visibility mode will work as expected if used directly.
(go into detail on the exact ones that dont work if you feel its necessary, tho I kinda dont think its needed)
```

Reply:
? ->

These <ins>`CAM_HUD_VISIBILITY_*`</ins> defines exist to clarify exactly which HUD <del>V</del><ins>v</ins>isibility modes <ins>`HUD_VISIBILITY_*`</ins> are supported by the camera system. While most of them map 1 to 1 with their HUD visibility counterparts, not all <del>values for</del> HUD visibility mode <ins>values `HUD_VISIBILITY_*`</ins> will work as expected if used directly.

Notably:
- `CAM_HUD_VISIBILITY_ALL` (0) maps to `HUD_VISIBILITY_ALL` (50), not `HUD_VISIBILITY_NO_CHANGE` (0)
- `HUD_VISIBILITY_NOTHING_INSTANT` (52) has no `CAM_HUD_VISIBILITY_*` mapping, because camera HUD visibility values are restriced to the 0-0xF range

Reply:
Comment has been adjusted.

Reply:
personally dont think we need to interject with `HUD_VISIBILITY_*` multiple times, and it makes it harder for me to read as one fluid sentence, but yeah just a preference thing

Reply:
(also I had `HUD visibility mode`  capitalized as `HUD Visibility Mode` in my first suggestion cause its a name of a thing, but I dont care too much either way)

Reply:
Makes sense, thanks for clarifying. Yeah that's fine with me